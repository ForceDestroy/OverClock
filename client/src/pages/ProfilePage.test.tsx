import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import axios from 'axios';
import { MemoryRouter } from 'react-router-dom';
import { describe, it, vi } from 'vitest';

import { App } from '../App';
import AuthTokenHelper from '../helpers/AuthTokenHelper';

vi.mock('axios');
vi.mock('../helpers/AuthTokenHelper');

describe('ProfilePage', () => {
  afterEach(() => {
    vi.clearAllMocks();
    vi.resetAllMocks();
  });
  it('Renders profile page to match snapshot', async () => {
    // Mock
    (axios.request as jest.Mock).mockReturnValue({
      data: {
        id: 'WEB_000000',
        name: 'web test',
        email: 'webtest@gmail.com',
        password: 'securepassword',
        address: '31 Jump Street',
        birthday: '0001-01-01T00:00:00',
        phoneNumber: 123,
        themeColor: 0,
        newPassword: '',
        confirmPassword: '',
      },
      status: 200,
      statusText: 'OK',
    });
    (AuthTokenHelper.validateToken as jest.Mock).mockReturnValue(
      'VerySecretToken'
    );

    // Arrange
    const snap = render(
      <MemoryRouter initialEntries={['/ProfilePage']}>
        <App />
      </MemoryRouter>
    );

    // Assert
    expect(snap).toMatchSnapshot();
  });
  it('Renders profile page after editing all fields', async () => {
    // Mock
    (axios.request as jest.Mock).mockReturnValue({
      data: {
        id: 'WEB_000000',
        name: 'web test',
        email: 'webtest@gmail.com',
        password: 'securepassword',
        address: '31 Jump Street',
        birthday: '0001-01-01T00:00:00',
        phoneNumber: 123,
        themeColor: 0,
        newPassword: '',
        confirmPassword: '',
      },
      status: 200,
      statusText: 'OK',
    });
    (AuthTokenHelper.validateToken as jest.Mock).mockReturnValue(
      'VerySecretToken'
    );

    // Arrange
    const user = userEvent.setup();
    render(
      <MemoryRouter initialEntries={['/ProfilePage']}>
        <App />
      </MemoryRouter>
    );

    // Act
    const nameField = screen.getByTestId('profileName');
    await user.type(nameField, ' web');
    const dobField = screen.getByTestId('profileBirthday');
    await user.type(dobField, '0001-01-01T00:00:00');
    const emailField = screen.getByTestId('profileEmail');
    await user.type(emailField, 'web');
    const phoneField = screen.getByTestId('profilePhoneNumber');
    await user.type(phoneField, '123');
    const addressField = screen.getByTestId('profileAddress');
    await user.type(addressField, ' Street');
    const newPasswordField = screen.getByTestId('profileNewPassword');
    await user.type(newPasswordField, 'newpassword');
    const confirmPasswordField = screen.getByTestId('profileConfirmPassword');
    await user.type(confirmPasswordField, 'newpassword');

    const saveButton = screen.getByTestId('saveButton');
    await user.click(saveButton);
    // Assert
    await waitFor(() => {
      expect(screen.getByText('web test web')).toBeInTheDocument();
      expect(screen.getByTestId('profileName')).toBeInTheDocument();
      expect(screen.getByTestId('profileEmail')).toBeInTheDocument();
      expect(screen.getByTestId('profileUserID')).toBeInTheDocument();
      expect(screen.getByTestId('profilePhoneNumber')).toBeInTheDocument();
      expect(screen.getByTestId('profileBirthday')).toBeInTheDocument();
      expect(screen.getByTestId('profileAddress')).toBeInTheDocument();
      expect(screen.getByTestId('profilePassword')).toBeInTheDocument();
      expect(screen.getByTestId('profileNewPassword')).toBeInTheDocument();
      expect(screen.getByTestId('profileConfirmPassword')).toBeInTheDocument();
      expect(
        screen.queryByText('Passwords do not match')
      ).not.toBeInTheDocument();
    });
  });

  it('Renders profile page after editing a field and cancelling', async () => {
    // Mock
    (axios.request as jest.Mock).mockReturnValue({
      data: {
        id: 'WEB_000000',
        name: 'web test',
        email: 'webtest@gmail.com',
        password: 'securepassword',
        address: '31 Jump Street',
        birthday: '0001-01-01T00:00:00',
        phoneNumber: 123,
        themeColor: 0,
        newPassword: '',
        confirmPassword: '',
      },
      status: 200,
      statusText: 'OK',
    });
    (AuthTokenHelper.validateToken as jest.Mock).mockReturnValue(
      'VerySecretToken'
    );

    // Arrange
    const user = userEvent.setup();
    render(
      <MemoryRouter initialEntries={['/ProfilePage']}>
        <App />
      </MemoryRouter>
    );

    // Act
    const nameField = screen.getByTestId('profileName');
    await user.type(nameField, 'web web');

    const cancelButton = screen.getByTestId('cancelButton');
    await user.click(cancelButton);
    // Assert
    await waitFor(() => {
      expect(screen.getByText('web test')).toBeInTheDocument();
    });
  });
  it('Renders profile page after entering mismatched passwords', async () => {
    // Mock
    (axios.request as jest.Mock).mockReturnValue({
      data: {
        id: 'WEB_000000',
        name: 'web test',
        email: 'webtest@gmail.com',
        password: 'securepassword',
        address: '31 Jump Street',
        birthday: '0001-01-01T00:00:00',
        phoneNumber: 123,
        themeColor: 0,
        newPassword: '',
        confirmPassword: '',
      },
      status: 200,
      statusText: 'OK',
    });
    (AuthTokenHelper.validateToken as jest.Mock).mockReturnValue(
      'VerySecretToken'
    );

    // Arrange
    const user = userEvent.setup();
    render(
      <MemoryRouter initialEntries={['/ProfilePage']}>
        <App />
      </MemoryRouter>
    );

    // Act
    const newPasswordField = screen.getByTestId('profileNewPassword');
    await user.type(newPasswordField, 'newpassword');
    const confirmPasswordField = screen.getByTestId('profileConfirmPassword');
    await user.type(confirmPasswordField, 'incorrect');

    const saveButton = screen.getByTestId('saveButton');
    await user.click(saveButton);
    // Assert
    await waitFor(() => {
      expect(screen.getByText('Passwords do not match')).toBeInTheDocument();
    });
  });
});
