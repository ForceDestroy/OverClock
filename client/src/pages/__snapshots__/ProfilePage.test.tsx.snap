// Vitest Snapshot v1

exports[`ProfilePage > Renders profile page to match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <div
          class="StyledGrommet-sc-19lkkz7-0 jGVJhU"
        >
          <div>
            <div
              class="StyledGrid-sc-1wofa1l-0 bySYOJ"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 dDSiKk"
              >
                <div
                  class="StyledBox-sc-13pk1d4-0 ibTKOe"
                >
                  <a
                    aria-label="home"
                    class="StyledAnchor-sc-1rp7lwl-0 iUqWmS"
                    data-testid="home"
                    href="/"
                  >
                    <div
                      class="StyledBox-sc-13pk1d4-0 iNDmPr"
                    >
                      <span
                        class="StyledText-sc-1sadyjn-0 cMFmcc"
                      >
                         
                        Home
                         
                      </span>
                    </div>
                  </a>
                  <div
                    class="StyledBox__StyledBoxGap-sc-13pk1d4-1 eYHYac"
                  />
                  <a
                    aria-label="Calendar"
                    class="StyledAnchor-sc-1rp7lwl-0 iUqWmS"
                    data-testid="Calendar"
                    href="/EmployeeDashboard/Schedule"
                  >
                    <div
                      class="StyledBox-sc-13pk1d4-0 iNDmPr"
                    >
                      <span
                        class="StyledText-sc-1sadyjn-0 cMFmcc"
                      >
                         
                        Calendar
                         
                      </span>
                    </div>
                  </a>
                  <div
                    class="StyledBox__StyledBoxGap-sc-13pk1d4-1 eYHYac"
                  />
                  <a
                    aria-label="MyJob"
                    class="StyledAnchor-sc-1rp7lwl-0 iUqWmS"
                    data-testid="MyJob"
                    href="/"
                  >
                    <div
                      class="StyledBox-sc-13pk1d4-0 iNDmPr"
                    >
                      <span
                        class="StyledText-sc-1sadyjn-0 cMFmcc"
                      >
                         
                        My Job
                         
                      </span>
                    </div>
                  </a>
                </div>
                <div
                  class="StyledBox-sc-13pk1d4-0 bqfCTq"
                >
                  <div
                    class="StyledBox-sc-13pk1d4-0 iNDmPr"
                  >
                    <span
                      class="StyledText-sc-1sadyjn-0 cdHUSp"
                    >
                       
                       
                    </span>
                    <span
                      class="StyledText-sc-1sadyjn-0 hpCRgD"
                    >
                       Sales Associate 
                    </span>
                  </div>
                  <div
                    class="StyledBox__StyledBoxGap-sc-13pk1d4-1 lbllaT"
                  />
                  <a
                    class="StyledAnchor-sc-1rp7lwl-0 iUqWmS"
                    href="/ProfilePage"
                  >
                    <div
                      class="StyledBox-sc-13pk1d4-0 kobaLM"
                    >
                      <div
                        class="StyledBox-sc-13pk1d4-0 ebPCjN StyledAvatar-sc-1suyamb-1"
                      />
                    </div>
                  </a>
                  <div
                    class="StyledBox__StyledBoxGap-sc-13pk1d4-1 lbllaT"
                  />
                  <button
                    class="StyledButton-sc-323bzc-0 kYtlmw"
                    data-testid="logout"
                    type="button"
                  >
                    <svg
                      aria-label="Logout"
                      class="StyledIcon-sc-ofa7kd-0 cnewlt"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M13 9V2H1v20h12v-7m9-3H5m12-5 5 5-5 5"
                        fill="none"
                        stroke="#231F20"
                        stroke-width="2"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="StyledBox-sc-13pk1d4-0 eETalo"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 iNDmPr"
            >
              <div
                class="StyledGrid-sc-1wofa1l-0 OnxGO"
              >
                <div
                  class="StyledBox-sc-13pk1d4-0 gBGvlp"
                >
                  <div
                    class="StyledBox-sc-13pk1d4-0 vvPtR"
                  >
                    <div
                      class="StyledBox-sc-13pk1d4-0 ebPCjN StyledAvatar-sc-1suyamb-1"
                    />
                  </div>
                  <div
                    class="StyledBox__StyledBoxGap-sc-13pk1d4-1 qdAYv"
                  />
                  <div
                    class="StyledBox-sc-13pk1d4-0 fkPXRa"
                  >
                    <span
                      class="StyledText-sc-1sadyjn-0 eHAjbp"
                    />
                    <br />
                    <span
                      class="StyledText-sc-1sadyjn-0 etAqJt"
                    >
                      Sales Associate
                    </span>
                    <br />
                    <br />
                  </div>
                </div>
                <div
                  class="StyledBox-sc-13pk1d4-0 bUHLVK"
                >
                  <div
                    class="StyledBox-sc-13pk1d4-0 gKOwbA"
                  >
                    <span
                      class="StyledText-sc-1sadyjn-0 kePuJz"
                    >
                      General Information
                    </span>
                    <br />
                    <div
                      class="StyledBox-sc-13pk1d4-0 jzOfgr"
                    >
                      <div
                        class="StyledBox-sc-13pk1d4-0 gUZObb"
                      >
                        <span
                          class="StyledText-sc-1sadyjn-0 hpCRgD"
                        >
                           Name 
                        </span>
                        <div
                          class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kZzThM"
                        >
                          <input
                            autocomplete="off"
                            class="StyledTextInput-sc-1x30a0s-0 jxFxho"
                            data-testid="profileName"
                            value=""
                          />
                        </div>
                      </div>
                    </div>
                    <br />
                    <div
                      class="StyledBox-sc-13pk1d4-0 gUZObb"
                    >
                      <span
                        class="StyledText-sc-1sadyjn-0 hpCRgD"
                      >
                         Id 
                      </span>
                      <div
                        class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kZzThM"
                      >
                        <input
                          autocomplete="off"
                          class="StyledTextInput-sc-1x30a0s-0 BKoXR"
                          data-testid="profileUserID"
                          disabled=""
                          value=""
                        />
                      </div>
                    </div>
                    <br />
                    <div
                      class="StyledBox-sc-13pk1d4-0 gUZObb"
                    >
                      <span
                        class="StyledText-sc-1sadyjn-0 hpCRgD"
                      >
                         Date of Birth 
                      </span>
                      <div
                        class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kZzThM"
                      >
                        <input
                          autocomplete="off"
                          class="StyledTextInput-sc-1x30a0s-0 jxFxho"
                          data-testid="profileBirthday"
                          placeholder="YYYY-MM-DD"
                          value=""
                        />
                      </div>
                    </div>
                    <br />
                    <div
                      class="StyledBox-sc-13pk1d4-0 gUZObb"
                    >
                      <span
                        class="StyledText-sc-1sadyjn-0 hpCRgD"
                      >
                         Email 
                      </span>
                      <div
                        class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kZzThM"
                      >
                        <input
                          autocomplete="off"
                          class="StyledTextInput-sc-1x30a0s-0 jxFxho"
                          data-testid="profileEmail"
                          value=""
                        />
                      </div>
                    </div>
                    <br />
                    <div
                      class="StyledBox-sc-13pk1d4-0 jzOfgr"
                    >
                      <div
                        class="StyledBox-sc-13pk1d4-0 gUZObb"
                      >
                        <span
                          class="StyledText-sc-1sadyjn-0 hpCRgD"
                        >
                           Phone Number 
                        </span>
                        <div
                          class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kZzThM"
                        >
                          <input
                            autocomplete="off"
                            class="StyledTextInput-sc-1x30a0s-0 jxFxho"
                            data-testid="profilePhoneNumber"
                            value="0"
                          />
                        </div>
                      </div>
                      <div
                        class="StyledBox__StyledBoxGap-sc-13pk1d4-1 lbllaT"
                      />
                      <div
                        class="StyledBox-sc-13pk1d4-0 gUZObb"
                      >
                        <span
                          class="StyledText-sc-1sadyjn-0 hpCRgD"
                        >
                           Address 
                        </span>
                        <div
                          class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kZzThM"
                        >
                          <input
                            autocomplete="off"
                            class="StyledTextInput-sc-1x30a0s-0 jxFxho"
                            data-testid="profileAddress"
                            value=""
                          />
                        </div>
                      </div>
                    </div>
                    <br />
                    <span
                      class="StyledText-sc-1sadyjn-0 kePuJz"
                    >
                      Password
                    </span>
                    <div
                      class="StyledBox-sc-13pk1d4-0 jzOfgr"
                    >
                      <div
                        class="StyledBox-sc-13pk1d4-0 gUZObb"
                      >
                        <span
                          class="StyledText-sc-1sadyjn-0 hpCRgD"
                        >
                           Old Password 
                        </span>
                        <div
                          class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kZzThM"
                        >
                          <input
                            autocomplete="off"
                            class="StyledTextInput-sc-1x30a0s-0 BKoXR"
                            data-testid="profilePassword"
                            disabled=""
                            value="************"
                          />
                        </div>
                      </div>
                      <div
                        class="StyledBox__StyledBoxGap-sc-13pk1d4-1 lbllaT"
                      />
                      <div
                        class="StyledBox-sc-13pk1d4-0 gUZObb"
                      >
                        <span
                          class="StyledText-sc-1sadyjn-0 hpCRgD"
                        >
                           New Password 
                        </span>
                        <div
                          class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kZzThM"
                        >
                          <input
                            autocomplete="off"
                            class="StyledTextInput-sc-1x30a0s-0 jxFxho"
                            data-testid="profileNewPassword"
                            value=""
                          />
                        </div>
                      </div>
                      <div
                        class="StyledBox__StyledBoxGap-sc-13pk1d4-1 lbllaT"
                      />
                      <div
                        class="StyledBox-sc-13pk1d4-0 gUZObb"
                      >
                        <span
                          class="StyledText-sc-1sadyjn-0 hpCRgD"
                        >
                           Confirm New Password 
                        </span>
                        <div
                          class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kZzThM"
                        >
                          <input
                            autocomplete="off"
                            class="StyledTextInput-sc-1x30a0s-0 jxFxho"
                            data-testid="profileConfirmPassword"
                            value=""
                          />
                        </div>
                      </div>
                    </div>
                    <div
                      class="StyledBox-sc-13pk1d4-0 lihpAm"
                    >
                      <div
                        class="StyledBox-sc-13pk1d4-0 jzOfgr"
                      >
                        <button
                          class="StyledButton-sc-323bzc-0 hiWqMi"
                          data-testid="cancelButton"
                          type="button"
                        >
                          <div
                            class="StyledBox-sc-13pk1d4-0 hZQWMp"
                          >
                            <br />
                            <span
                              class="StyledText-sc-1sadyjn-0 oDQrS"
                            >
                              Cancel
                            </span>
                            <br />
                          </div>
                        </button>
                        <div
                          class="StyledBox__StyledBoxGap-sc-13pk1d4-1 gJrKvj"
                        />
                        <button
                          class="StyledButton-sc-323bzc-0 fHvKkw"
                          data-testid="saveButton"
                          type="button"
                        >
                          <div
                            class="StyledBox-sc-13pk1d4-0 dfdKQJ"
                          >
                            <br />
                            <span
                              class="StyledText-sc-1sadyjn-0 cKTRLX"
                            >
                              Save Changes
                            </span>
                            <br />
                          </div>
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <div
        class="StyledGrommet-sc-19lkkz7-0 jGVJhU"
      >
        <div>
          <div
            class="StyledGrid-sc-1wofa1l-0 bySYOJ"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 dDSiKk"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 ibTKOe"
              >
                <a
                  aria-label="home"
                  class="StyledAnchor-sc-1rp7lwl-0 iUqWmS"
                  data-testid="home"
                  href="/"
                >
                  <div
                    class="StyledBox-sc-13pk1d4-0 iNDmPr"
                  >
                    <span
                      class="StyledText-sc-1sadyjn-0 cMFmcc"
                    >
                       
                      Home
                       
                    </span>
                  </div>
                </a>
                <div
                  class="StyledBox__StyledBoxGap-sc-13pk1d4-1 eYHYac"
                />
                <a
                  aria-label="Calendar"
                  class="StyledAnchor-sc-1rp7lwl-0 iUqWmS"
                  data-testid="Calendar"
                  href="/EmployeeDashboard/Schedule"
                >
                  <div
                    class="StyledBox-sc-13pk1d4-0 iNDmPr"
                  >
                    <span
                      class="StyledText-sc-1sadyjn-0 cMFmcc"
                    >
                       
                      Calendar
                       
                    </span>
                  </div>
                </a>
                <div
                  class="StyledBox__StyledBoxGap-sc-13pk1d4-1 eYHYac"
                />
                <a
                  aria-label="MyJob"
                  class="StyledAnchor-sc-1rp7lwl-0 iUqWmS"
                  data-testid="MyJob"
                  href="/"
                >
                  <div
                    class="StyledBox-sc-13pk1d4-0 iNDmPr"
                  >
                    <span
                      class="StyledText-sc-1sadyjn-0 cMFmcc"
                    >
                       
                      My Job
                       
                    </span>
                  </div>
                </a>
              </div>
              <div
                class="StyledBox-sc-13pk1d4-0 bqfCTq"
              >
                <div
                  class="StyledBox-sc-13pk1d4-0 iNDmPr"
                >
                  <span
                    class="StyledText-sc-1sadyjn-0 cdHUSp"
                  >
                     
                     
                  </span>
                  <span
                    class="StyledText-sc-1sadyjn-0 hpCRgD"
                  >
                     Sales Associate 
                  </span>
                </div>
                <div
                  class="StyledBox__StyledBoxGap-sc-13pk1d4-1 lbllaT"
                />
                <a
                  class="StyledAnchor-sc-1rp7lwl-0 iUqWmS"
                  href="/ProfilePage"
                >
                  <div
                    class="StyledBox-sc-13pk1d4-0 kobaLM"
                  >
                    <div
                      class="StyledBox-sc-13pk1d4-0 ebPCjN StyledAvatar-sc-1suyamb-1"
                    />
                  </div>
                </a>
                <div
                  class="StyledBox__StyledBoxGap-sc-13pk1d4-1 lbllaT"
                />
                <button
                  class="StyledButton-sc-323bzc-0 kYtlmw"
                  data-testid="logout"
                  type="button"
                >
                  <svg
                    aria-label="Logout"
                    class="StyledIcon-sc-ofa7kd-0 cnewlt"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M13 9V2H1v20h12v-7m9-3H5m12-5 5 5-5 5"
                      fill="none"
                      stroke="#231F20"
                      stroke-width="2"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 eETalo"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 iNDmPr"
          >
            <div
              class="StyledGrid-sc-1wofa1l-0 OnxGO"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 gBGvlp"
              >
                <div
                  class="StyledBox-sc-13pk1d4-0 vvPtR"
                >
                  <div
                    class="StyledBox-sc-13pk1d4-0 ebPCjN StyledAvatar-sc-1suyamb-1"
                  />
                </div>
                <div
                  class="StyledBox__StyledBoxGap-sc-13pk1d4-1 qdAYv"
                />
                <div
                  class="StyledBox-sc-13pk1d4-0 fkPXRa"
                >
                  <span
                    class="StyledText-sc-1sadyjn-0 eHAjbp"
                  />
                  <br />
                  <span
                    class="StyledText-sc-1sadyjn-0 etAqJt"
                  >
                    Sales Associate
                  </span>
                  <br />
                  <br />
                </div>
              </div>
              <div
                class="StyledBox-sc-13pk1d4-0 bUHLVK"
              >
                <div
                  class="StyledBox-sc-13pk1d4-0 gKOwbA"
                >
                  <span
                    class="StyledText-sc-1sadyjn-0 kePuJz"
                  >
                    General Information
                  </span>
                  <br />
                  <div
                    class="StyledBox-sc-13pk1d4-0 jzOfgr"
                  >
                    <div
                      class="StyledBox-sc-13pk1d4-0 gUZObb"
                    >
                      <span
                        class="StyledText-sc-1sadyjn-0 hpCRgD"
                      >
                         Name 
                      </span>
                      <div
                        class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kZzThM"
                      >
                        <input
                          autocomplete="off"
                          class="StyledTextInput-sc-1x30a0s-0 jxFxho"
                          data-testid="profileName"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                  <br />
                  <div
                    class="StyledBox-sc-13pk1d4-0 gUZObb"
                  >
                    <span
                      class="StyledText-sc-1sadyjn-0 hpCRgD"
                    >
                       Id 
                    </span>
                    <div
                      class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kZzThM"
                    >
                      <input
                        autocomplete="off"
                        class="StyledTextInput-sc-1x30a0s-0 BKoXR"
                        data-testid="profileUserID"
                        disabled=""
                        value=""
                      />
                    </div>
                  </div>
                  <br />
                  <div
                    class="StyledBox-sc-13pk1d4-0 gUZObb"
                  >
                    <span
                      class="StyledText-sc-1sadyjn-0 hpCRgD"
                    >
                       Date of Birth 
                    </span>
                    <div
                      class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kZzThM"
                    >
                      <input
                        autocomplete="off"
                        class="StyledTextInput-sc-1x30a0s-0 jxFxho"
                        data-testid="profileBirthday"
                        placeholder="YYYY-MM-DD"
                        value=""
                      />
                    </div>
                  </div>
                  <br />
                  <div
                    class="StyledBox-sc-13pk1d4-0 gUZObb"
                  >
                    <span
                      class="StyledText-sc-1sadyjn-0 hpCRgD"
                    >
                       Email 
                    </span>
                    <div
                      class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kZzThM"
                    >
                      <input
                        autocomplete="off"
                        class="StyledTextInput-sc-1x30a0s-0 jxFxho"
                        data-testid="profileEmail"
                        value=""
                      />
                    </div>
                  </div>
                  <br />
                  <div
                    class="StyledBox-sc-13pk1d4-0 jzOfgr"
                  >
                    <div
                      class="StyledBox-sc-13pk1d4-0 gUZObb"
                    >
                      <span
                        class="StyledText-sc-1sadyjn-0 hpCRgD"
                      >
                         Phone Number 
                      </span>
                      <div
                        class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kZzThM"
                      >
                        <input
                          autocomplete="off"
                          class="StyledTextInput-sc-1x30a0s-0 jxFxho"
                          data-testid="profilePhoneNumber"
                          value="0"
                        />
                      </div>
                    </div>
                    <div
                      class="StyledBox__StyledBoxGap-sc-13pk1d4-1 lbllaT"
                    />
                    <div
                      class="StyledBox-sc-13pk1d4-0 gUZObb"
                    >
                      <span
                        class="StyledText-sc-1sadyjn-0 hpCRgD"
                      >
                         Address 
                      </span>
                      <div
                        class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kZzThM"
                      >
                        <input
                          autocomplete="off"
                          class="StyledTextInput-sc-1x30a0s-0 jxFxho"
                          data-testid="profileAddress"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                  <br />
                  <span
                    class="StyledText-sc-1sadyjn-0 kePuJz"
                  >
                    Password
                  </span>
                  <div
                    class="StyledBox-sc-13pk1d4-0 jzOfgr"
                  >
                    <div
                      class="StyledBox-sc-13pk1d4-0 gUZObb"
                    >
                      <span
                        class="StyledText-sc-1sadyjn-0 hpCRgD"
                      >
                         Old Password 
                      </span>
                      <div
                        class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kZzThM"
                      >
                        <input
                          autocomplete="off"
                          class="StyledTextInput-sc-1x30a0s-0 BKoXR"
                          data-testid="profilePassword"
                          disabled=""
                          value="************"
                        />
                      </div>
                    </div>
                    <div
                      class="StyledBox__StyledBoxGap-sc-13pk1d4-1 lbllaT"
                    />
                    <div
                      class="StyledBox-sc-13pk1d4-0 gUZObb"
                    >
                      <span
                        class="StyledText-sc-1sadyjn-0 hpCRgD"
                      >
                         New Password 
                      </span>
                      <div
                        class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kZzThM"
                      >
                        <input
                          autocomplete="off"
                          class="StyledTextInput-sc-1x30a0s-0 jxFxho"
                          data-testid="profileNewPassword"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      class="StyledBox__StyledBoxGap-sc-13pk1d4-1 lbllaT"
                    />
                    <div
                      class="StyledBox-sc-13pk1d4-0 gUZObb"
                    >
                      <span
                        class="StyledText-sc-1sadyjn-0 hpCRgD"
                      >
                         Confirm New Password 
                      </span>
                      <div
                        class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 kZzThM"
                      >
                        <input
                          autocomplete="off"
                          class="StyledTextInput-sc-1x30a0s-0 jxFxho"
                          data-testid="profileConfirmPassword"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="StyledBox-sc-13pk1d4-0 lihpAm"
                  >
                    <div
                      class="StyledBox-sc-13pk1d4-0 jzOfgr"
                    >
                      <button
                        class="StyledButton-sc-323bzc-0 hiWqMi"
                        data-testid="cancelButton"
                        type="button"
                      >
                        <div
                          class="StyledBox-sc-13pk1d4-0 hZQWMp"
                        >
                          <br />
                          <span
                            class="StyledText-sc-1sadyjn-0 oDQrS"
                          >
                            Cancel
                          </span>
                          <br />
                        </div>
                      </button>
                      <div
                        class="StyledBox__StyledBoxGap-sc-13pk1d4-1 gJrKvj"
                      />
                      <button
                        class="StyledButton-sc-323bzc-0 fHvKkw"
                        data-testid="saveButton"
                        type="button"
                      >
                        <div
                          class="StyledBox-sc-13pk1d4-0 dfdKQJ"
                        >
                          <br />
                          <span
                            class="StyledText-sc-1sadyjn-0 cKTRLX"
                          >
                            Save Changes
                          </span>
                          <br />
                        </div>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
